src/workerd/api/sql.h

```diff
@@ -107,6 +107,58 @@ class SqlStorage final: public jsg::Object, private SqliteDatabase::Regulator {
     kj::Maybe<kj::Array<jsg::JsRef<jsg::JsString>>> names;
   };
 
+  // A statement in the statement cache.
+  struct CachedStatement: public kj::Refcounted {
+    jsg::HashableV8Ref<v8::String> query;
+    size_t statementSize;
+    SqliteDatabase::Statement statement;
+    CachedColumnNames cachedColumnNames;
+    kj::ListLink<CachedStatement> lruLink;
+
+    CachedStatement(jsg::Lock& js,
+        SqlStorage& sqlStorage,
+        SqliteDatabase& db,
+        jsg::JsString jsQuery,
+        kj::String kjQuery)
+        : query(js.v8Isolate, jsQuery),
+          statementSize(kjQuery.size()),
+          statement(db.prepareMulti(sqlStorage, kj::mv(kjQuery))) {}
+  };
+
+  class StatementCacheCallbacks {
+  public:
+    inline const jsg::HashableV8Ref<v8::String>& keyForRow(
+        const kj::Rc<CachedStatement>& entry) const {
+      return entry->query;
+    }
+
+    inline bool matches(const kj::Rc<CachedStatement>& entry, jsg::JsString key) const {
+      return entry->query == key;
+    }
+    inline bool matches(
+        const kj::Rc<CachedStatement>& entry, const jsg::HashableV8Ref<v8::String>& key) const {
+      return entry->query == key;
+    }
+
+    inline auto hashCode(jsg::JsString key) const {
+      return key.hashCode();
+    }
+    inline auto hashCode(const jsg::HashableV8Ref<v8::String>& key) const {
+      return key.hashCode();
+    }
+  };
+
+  using StatementMap = kj::Table<kj::Rc<CachedStatement>, kj::HashIndex<StatementCacheCallbacks>>;
```

Comment:
It's because I wanted the key to be stored as a member of `CachedStatement` which is held by `kj::Rc`, not inlined into the table. `HashMap` doesn't quite have that structure. (I'll add a comment.)